tests/test_intercept.js

```diff
@@ -2297,6 +2281,39 @@ test('pending mocks works', function(t) {
   });
 });
 
+test('pending mocks doesn\'t include optional mocks', function(t) {
+  var scope = nock('http://example.com')
+    .get('/nonexistent')
+    .optionally()
+    .reply(200);
+
+  t.deepEqual(nock.pendingMocks(), []);
+  t.end();
+});
+
+test('optional mocks are still functional', function(t) {
+  var scope = nock('http://example.com')
+    .get('/abc')
+    .optionally()
+    .reply(200);
+
+  var req = http.get({host: 'example.com', path: '/abc'}, function(res) {
+    t.assert(res.statusCode === 200, "should still mock requests");
+    t.deepEqual(nock.pendingMocks(), []);
+    t.end();
+  });
+});
+
+test('isDone is true with optional mocks outstanding', function(t) {
+  var scope = nock('http://example.com')
+    .get('/abc')
+    .optionally()
+    .reply(200);
+
+  t.ok(scope.isDone());
+  t.end();
+});
```

Comment:
With the code in this PR, I would expect a mock that's both `optional` and `persist` to be absent from `pendingMocks` but intercept requests multiple times.

Something analogous could be said about `optional` and `times(n)`.

Would adding tests for these combined behaviours be an overkill?